api_test.go

```diff
@@ -37,6 +27,12 @@ func WithServer(t *testing.T, f func(string)) {
 		time.Sleep(50 * time.Millisecond)
 	}
 
+	// Need to assign here since we need to call NewProxy to set the client.
+	testProxy.Name = "mysql_master"
+	testProxy.Listen = "localhost:3310"
+	testProxy.Upstream = "localhost:20001"
+	testProxy.Enabled = true
```

Comment:
Maybe pass in the name and upstream address to `NewProxy` and then the listen address defaults to :0 and enabled = true